### PR TITLE
Update multicolumnlistbox

### PR DESCRIPTION
**Pull request by edoffagne**
Issues fixed by this PR
-----------------------
Add vertical separators to multicolumnlistbox #223

What does this implement/fix?
-----------------------------
Added optional parameter in MultiColumnListBox that specifies the separator character to be used.

Any other comments?
-------------------
Feedback and comments welcome :-) thank you

### CODE REVIEW

File: asciimatics/widgets.py [hunk 1]
```diff
@@ -2453,7 +2453,7 @@ class _BaseListBox(with_metaclass(ABCMeta, Widget)):
                  "_on_select", "_validator", "_search", "_last_search", "_scroll_bar", "_parser"]
 
     def __init__(self, height, options, titles=None, label=None, name=None, parser=None,
-                 on_change=None, on_select=None, validator=None):
+                 on_change=None, on_select=None, validator=None, space_delimiter=None):
```
Comment (peterbrittain) [inline asciimatics/widgets.py]: Do you need this change?  Looks like you only use the spacing delimiter in the child class and so this is unecessary.

File: tests/test_widgets.py [hunk 2]
```diff
@@ -1129,6 +1129,135 @@ def test_multi_column_list_box(self):
...
+            titles=["A", "B", "C", "D", "E", "F"],
+            name="mc_list",
+            space_delimiter='|')
+        form.add_layout(layout)
+        layout.add_widget(mc_list)
+        form.fix()
+        form.register_scene(scene)
+        form.reset()
+
+        # Check we have a default value for our list.
```
Comment (peterbrittain) [inline tests/test_widgets.py]: This clearly tests a lot more than just the new delimiter... Can you please remove all the other tests and just leave the stuff for your new function?

Comment (edoffagne) [reply tests/test_widgets.py]: Done.

File: asciimatics/widgets.py (hunk 1 above)
Comment (edoffagne) [reply asciimatics/widgets.py]: Indeed you are right. I have updated the code accordingly. Thanks

**Comment by peterbrittain** [comment]
Many thanks!

**Pull request merged**